core/libp2p/protocol/kademlia/value_store.hpp

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_KADEMLIA_VALUE_STORE_HPP
+#define KAGOME_KADEMLIA_VALUE_STORE_HPP
+
+#include "libp2p/protocol/kademlia/common.hpp"
+
+namespace libp2p::protocol::kademlia {
+
+  /// ValueStore is a basic Put/Get interface.
+  struct ValueStore {
```

Comment:
Looks like the interface can be generalized to the templated one and be in `libp2p::basic`

Reply:
Maybe. Maybe not.